.github/workflows/release.yaml

```diff
@@ -0,0 +1,33 @@
+name: Release
+on:
+  release:
+    types: [published]
+
+jobs:
+  release:
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    if: github.repository == 'pytest-dev/pytest-timeout'
+
+    permissions:
+      id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
```

Comment:
I think the publish action recommends building without this permission to minimise access to this permission.  I guess that means adding a separate job for the building and passing the build output along as artefacts?

Reply:
If @aqeelat wants to go with that route, see how pytest-mock does it: https://github.com/pytest-dev/pytest-mock/blob/fcde66b13d09cb5507fe7e4f35e171adaf22994d/.github/workflows/deploy.yml#L13-L25

(we also build the package using @hynek's excellent `build-and-inspect-python-package` action).